### ISSUE TITLE
QGIS Python fails to setup environment on Windows

### ISSUE THREAD

**Issue by komima** [qgis/QGIS#43308]
**Describe the bug**

[setupenv in qgis python module](https://github.com/qgis/QGIS/blob/master/python/__init__.py#L57) fails to setup the environment correctly (Windows-specific method to read the vars from eg. `qgis-bin.env` file in the osgeo4w installation bin directory.

**How to Reproduce**

* Create new venv with the python wrapper bat (`python-qgis -m venv .venv --system-site-packages`).
* Activate the environment in another terminal (that did not just run the python wrapper bat) and run python console.
* Add QGIS python module directory to the search path with `sys.path.append('C:/the-osgeo-install-path/apps/qgis/python')`.

\>=3.16.7 (with python 3.9) cannot find plugins
* Run `from qgis.testing import setup_app` and `setup_app()` to initialize a new application, app starts with newlines in paths.
* Create a postgres provider vector layer with `testlayer = QgsVectorLayer("dbname=testing","testlayer","postgres")`
* Command fails with invalid provider message.

<3.16.7 (and 3.16.7 with python 3.7) this was causing qt env problems:
* Run `from qgis.testing import setup_app` and `setup_app()` to initialize a new application.
* Qt only searches the first part of QT_PLUGIN_PATH since the second path has a new line at the end, and app cannot start.

![image](https://user-images.githubusercontent.com/58747243/119032792-41421100-b9b5-11eb-8a3c-155f7c05bf3b.png)

When running with the python wrapper bat, QGIS_PREFIX_PATH gets set and setupenv does nothing on import.

![image](https://user-images.githubusercontent.com/58747243/119034418-f3c6a380-b9b6-11eb-882d-5827d63dc6b1.png)

When running with the venv python, QGIS_PREFIX_PATH is not set, setupenv runs and variables are added with the extra newlines.

![image](https://user-images.githubusercontent.com/58747243/119035514-52405180-b9b8-11eb-8fb4-d142c45b05d4.png)

**QGIS and OS versions**

3.16.7, 3.18.3, also 3.16.6 and 3.14.15 tested.

QGIS version | 3.14.15-Pi | QGIS code revision | d5114d2cfa
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | Windows 10 (10.0)
Active python plugins | db_manager; MetaSearch; processing

**Possible fix**

Use `data = linedata[1].strip()` on [this line](https://github.com/qgis/QGIS/blob/master/python/__init__.py#L60)?

**Comment by gioman** [comment]
> Up-to-date osgeo4w installs (3.18.3 and 3.16.7) are working

@komima closing then? The other issue must be reported in separate tickets.

**Comment by komima** [comment]
> > Up-to-date osgeo4w installs (3.18.3 and 3.16.7) are working
> 
> @komima closing then? The other issue must be reported in separate tickets.

I thinks its still the same bug (newlines in paths causing problems), just appears in a different way on current versions?

**Comment by komima** [comment]
Tested this more, fresh install with standalone installer or the network installer using qgis-ltr package version 3.16.7 installs with Python 3.7, which fails with Qt platform plugin error.

Adding ; at the end to QT_PLUGIN_PATH (or changing the order of the 2 paths) in qgis-ltr-bin.env solves the start_app error, but the newline in prefix path still causes provider plugins not to be found.

I can't see any reason why the ending newline should be in the paths when setting those to os.environ? I created a PR with the change.
